odbc/src/api/data.rs

```diff
@@ -1254,7 +1278,7 @@ pub mod isize_len {
         let (len, ret) =
             set_output_wstring_helper(message.get(index..).unwrap(), output_ptr, buffer_len);
         // the returned length should always be the total length of the data.
-        *text_length_ptr = (message.len() - index) as Len;
+        *text_length_ptr = (size_of::<WChar>() * (message.len() - index)) as Len;
```

Comment:
Is index here bytes or characters? The line following looks like it's working characters still?

 I think it might help to have a helper for the bytes methods that returns the bytes instead of len to make it clear that's what we're doing. Then we could do something like `(bytes, ret) = set_output_wstring_bytes_helper(...)`

Reply:
index is number of chars, otherwise we'd end up with units of bytes^2 

the multiplication here is index `<char> * size_of::WChar> <bytes / char>`

Reply:
Fair. In other methods we're using `size_of::<WChar>() * len` but in this we're using `message.len()`.

```suggestion
        *text_length_ptr = (size_of::<WChar>() * (len - index)) as Len;
```

Reply:
The initial code is correct, we need to use message.len() here. When streaming the data, we don't report how much data is in the buffer, but how much data is available for retrieval. My guess is that it is because until you get to the last call, the buffer will be full, so it does not matter, but users might want to know how many more calls they need to do.

Reply:
Yep, there is a comment there that says "the returned length should always be the total length of the data".

Initially, I used len. But when I reread the docs I found out that was wrong.

Reply:
I'm still unsure why we aren't using `len` here that's returned from the helper since it takes into account the possiblity of truncation.